Backend/templates/gitlab-ci/master-gitlab-ci.yml

```diff
@@ -0,0 +1,276 @@
+
+
+
+##################
+# Build tag jobs #
+##################
+
+DeployVersionDev: 
+  stage: BuildNumber
+  only:
+    - /builds/
+  except:
+    - branches
+  script:
+    - echo $CI_COMMIT_TAG | sed -n 's/builds\/\([0-9]\)/\1/p' > .build_number
+    - echo "Build number "$(cat .build_number)
+  artifacts:
+    paths:
+      #TODO maybe this should be a different name? 🤔
+      - .build_number 
+
+
+DeployDev:
+  extends: .deploy_template
+  variables:
+    PROJECT: "youclap-dev"
+  only:
+    - /builds/
+
+
+DocumentationDev:
+  extends: .documentation_template
+  only:
+    - /builds/
+  dependencies:
+    - DeployVersionDev 
+
+
+DeployDocumentationDev:
+  extends: .deploy_documentation_template
+  variables:
+    PROJECT: "youclap-dev"
+    DOC: "dev" #TODO find a better name
+  only:
+    - /builds/
+  dependencies:
+    - DeployVersionDev
+    - DocumentationDev
+
+
+#####################
+# Releases tag jobs #
+#####################
+
+DeployVersionProd: 
+  stage: BuildNumber
+  only:
+    - /releases/
+  except:
+    - branches
+  script:
+    - echo $CI_COMMIT_TAG | sed -n 's/releases\/\([0-9].[0-9].[0-9]\)/\1/p' > .build_number
+    - cat .build_number
+  artifacts:
+    paths:
+      - .build_number
+
+
+DeployProd:
+  extends: .deploy_template
+  only:
+    - /releases/
+  variables:
+    PROJECT: "youclap-prod"
+
+
+DocumentationProd:
+  extends: .documentation_template
+  only:
+    - /releases/
+  dependencies:
+    - DeployVersionProd
+
+
+DeployDocumentationProd:
+  extends: .deploy_documentation_template
+  variables:
+    PROJECT: "youclap-prod"
+    DOC: "prod"
+  only:
+    - /releases/
+  dependencies:
+    - DeployVersionProd
+    - DocumentationProd
+
+
+#############
+# Templates #
+#############
+
+.build_template:
+  stage: Build
+  only:
+    - master
+
+
+.deploy_template:
+  stage: Deploy
+  image: devth/helm:$HELM_DOCKER_TAG
+  except:
+    - branches
+  script:
+    - BUILD_NUMBER=`cat .build_number`
+    - echo "Build number $BUILD_NUMBER"
+
+    - gcloud auth activate-service-account --key-file=$GCLOUD_SERVICE_ACCOUNT_WS_KUBERNETES --project $PROJECT
+    - gcloud container clusters get-credentials gcloud-cluster-dev --zone europe-west1-d --project $PROJECT
+
+    - helm init --client-only
+
+    #TODO check if this is correct to have CHART_NAME here
+    - helm upgrade $CHART_NAME ./$CHART_NAME 
+      --install
+      --recreate-pods
+      --version $BUILD_NUMBER
+      --set image.repository=$CI_REGISTRY/$CI_PROJECT_PATH/builds
+      --set image.tag=$BUILD_NUMBER
+      --set image.pullSecret=gitlab-auth #TODO improve this 🔨
+
+
+.documentation_template:
+  image: openapitools/openapi-generator-cli:$OPEN_API_DOCKER_VERSION
+  stage: Documentation
+  except:
+    - branches
+  script:
+    - BUILD_NUMBERR=`cat .build_number`
+    - echo "Build number $BUILD_NUMBER"
+
+    - sed -i.bak 's/BUILD_NUMBER/'$build_number'/' swagger.yml # TODO maybe find a better way to do this 
```

Comment:
Should you also pass the `URL` of the service?

Reply:
not sure if it is worth 🤔